File: buildSrc/src/main/java/wolkenschloss/TransformationExtension.java
package wolkenschloss;

import org.gradle.api.file.DirectoryProperty;
import org.gradle.api.file.ProjectLayout;

public abstract class TransformationExtension {

    public void initialize(ProjectLayout layout) {
        getGeneratedCloudInitDirectory().set(layout.getBuildDirectory().dir("cloud-init"));
        getGeneratedVirshConfigDirectory().set(layout.getBuildDirectory().dir("config"));
        getSourceDirectory().set(layout.getProjectDirectory().dir("src"));
    }

    public abstract DirectoryProperty getSourceDirectory();

    public abstract DirectoryProperty getGeneratedCloudInitDirectory();

    public abstract DirectoryProperty getGeneratedVirshConfigDirectory();
}

File: buildSrc/src/main/kotlin/wolkenschloss/gradle/testbed/transformation/TransformationExtension.kt
package wolkenschloss.gradle.testbed.transformation

import org.gradle.api.file.DirectoryProperty
import org.gradle.api.file.ProjectLayout

abstract class TransformationExtension {
    fun initialize(layout: ProjectLayout) {
        generatedConfigurationDirectory.convention(layout.buildDirectory.dir("config"))
        sourceDirectory.set(layout.projectDirectory.dir("config"))
    }

    abstract val generatedConfigurationDirectory: DirectoryProperty
    abstract val sourceDirectory: DirectoryProperty
}